sites/main-site/src/content/blog/2025/refurbushing-the-pipeline-download.mdx

```diff
@@ -0,0 +1,85 @@
+---
+title: "Why are my pipeline download tests failing?"
+subtitle: "Refurbishing the pipeline downloads command"
+headerImage: "https://images.unsplash.com/photo-1636819488537-a9b1ffb315ce"
+headerImageAlt: "3D rendering of a cute little space rocket painted red and white with a fluffy smoke cloud coming out of it."
+pubDate: 2025-08-15T12:00:00+01:00
+authors:
+    - "ErikDanielsson"
+label:
+    - "tools"
+maxHeadingDepth: 3
+---
+
+import { Image } from "astro:assets";
+
+import old_nf_version from "@assets/images/blog/pipeline-download-refactor/old-nf-version.png";
+import rnaseq_in_action from "@assets/images/blog/pipeline-download-refactor/rnaseq-in-action.png";
+
+The `nf-core pipelines download` command, used when you want to run an `nf-core` pipeline in an offline compute environment, has recently undergone a [substantial refactor](https://github.com/nf-core/tools/pull/3634) that has recently been merged into the development version of `nf-core/tools`.
+The command has for a long time been reliant on complex regex logic to capture container strings in modules and config files.
+Due to the variability in where a pipeline can define containers, this has been prone to breaking.
```

Comment:
It would be good to add a paragraph with a bit of history on how the download command used to work befor, and what problems did we find.

Reply:
Sounds like a good idea!

Reply:
In case you need inspiration...I once made a meme about it.

Good examples of what can go wrong with regexing, [is the quote issue in the Bowtie ](https://github.com/nf-core/tools/issues/2392)module or r[ace conditions because Seqera containers are all called `/data`](https://github.com/nf-core/tools/issues/3285). 

<img width="2760" height="2100" alt="Fixing nf-core download" src="https://github.com/user-attachments/assets/7ddf0e52-6faf-4bba-b36d-ddd147ee9774" />

Reply:
Thanks! I'll link those PRs too if anyone is interested in learning more. 

Since you have maintained `nf-core download` for such a long time with the regex logic, feel free to update the blog post especially the command history as you see fit!